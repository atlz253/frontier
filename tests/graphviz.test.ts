import { Graph } from "graphlib";
import { write } from "graphlib-dot";
import { instance } from "@viz-js/viz";
import { describe, expect, test } from "vitest";

describe("graphviz", () => {
  test("JSON to DOT language", () => {
    const obj = {
      a: {},
      b: {
        dependencies: ["a"],
      },
      c: {
        dependencies: ["a", "b"],
      },
    };
    const graph = new Graph();
    for (const [name, options] of Object.entries(obj)) {
      graph.setNode(name);
      if ("dependencies" in options)
        options.dependencies.forEach((d) => graph.setEdge(name, d));
    }
    expect(write(graph)).toMatchSnapshot();
  });

  test("DOT language to SVG", async () => {
    const viz = await instance();
    const svg = viz.renderString(
      `
      digraph G {
        A -> B;
        B -> C;
        A -> C;
        C -> D;
      }
    `,
      { format: "svg" }
    );
    await expect(svg).toMatchFileSnapshot("__snapshots__/graphviz.test.ts.svg");
  });
});
