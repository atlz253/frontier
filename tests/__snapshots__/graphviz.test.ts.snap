// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`graphviz > JSON to DOT language 1`] = `
"strict digraph {
  a
  b
  c
  b -> a
  c -> a
  c -> b
}
"
`;
